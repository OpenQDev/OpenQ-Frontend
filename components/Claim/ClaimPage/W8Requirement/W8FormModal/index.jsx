import Image from 'next/image';
import Link from 'next/link';
import React from 'react';
import ModalLarge from '../../../../Utils/ModalLarge';

const index = ({ setShowModal }) => {
  const btn = (
    <button className='btn-primary' onClick={() => setShowModal(false)}>
      I understand
    </button>
  );
  return (
    <ModalLarge title={'W8/W9 Form'} footerRight={btn} setShowModal={setShowModal} resetState={setShowModal}>
      <div className='whitespace-pre-wrap px-8 py-2'>
        <p className='py-2'>
          Please complete and upload a form W-8. Choose one of five types, depending on your entity. We encourage you to
          consult with your own tax or financial adviser to determine which form is appropriate for you.
        </p>

        <br />

        <p className='flex py-2 gap-2'>
          1.
          <Image src={'/diverse/us-flag.jpg'} width={24} height={24} alt={'US-flag'} />
          W-9
        </p>

        <p className='py-2'>‚úÖ US Citizen or Resident</p>

        <p className='py-2'>‚úÖ Claiming as an individual</p>

        <p className='py-2'>
          Instructions:{' '}
          <Link
            href={'https://www.irs.gov/forms-pubs/about-form-w-9'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/forms-pubs/about-form-w-9
          </Link>
        </p>

        <p className='py-2'>
          Form:{' '}
          <Link
            href={'https://www.irs.gov/pub/irs-pdf/fw9.pdf'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/pub/irs-pdf/fw9.pdf
          </Link>
        </p>

        <br />

        <p className='py-2'>2. üåç W-8-BEN</p>

        <p className='py-2'>‚úÖ Not a US Citizen or Resident</p>

        <p className='py-2'>‚úÖ Claiming as an individual</p>

        <p className='py-2'>
          Instructions:{' '}
          <Link
            href={'https://www.irs.gov/instructions/iw8ben'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/instructions/iw8ben
          </Link>
        </p>

        <p className='py-2'>
          Form:{' '}
          <Link
            href={'https://www.irs.gov/pub/irs-pdf/fw8ben.pdf'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/pub/irs-pdf/fw8ben.pdf
          </Link>
        </p>

        <br />

        <p className='py-2'>3. üåé W-8-BEN-E</p>

        <p className='py-2'>‚úÖ Not a US Citizen or Resident</p>

        <p className='py-2'>‚úÖ Claiming as an entity</p>

        <p className='py-2'>
          Instructions:{' '}
          <Link
            href={'https://www.irs.gov/instructions/iw8bene'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/instructions/iw8bene
          </Link>
        </p>

        <p className='py-2'>
          Form:{' '}
          <Link
            href={'https://www.irs.gov/pub/irs-pdf/fw8bene.pdf'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/pub/irs-pdf/fw8bene.pdf
          </Link>
        </p>

        <br />

        <p className='py-2'>4. üåé W-8-ECI</p>

        <p className='py-2'>‚úÖ Not a US Citizen or Resident</p>

        <p className='py-2'>
          ‚úÖ Foreign Person‚Äôs Claim That Income is Effectively Connected With the Conduct of a Trade or Business in the
          United States
        </p>

        <p className='py-2'>
          Instructions:{' '}
          <Link
            href={'https://www.irs.gov/pub/irs-pdf/iw8eci.pdf'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/pub/irs-pdf/iw8eci.pdf
          </Link>
        </p>

        <p className='py-2'>
          Form:{' '}
          <Link
            href={'https://www.irs.gov/pub/irs-pdf/fw8eci.pdf'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/pub/irs-pdf/fw8eci.pdf
          </Link>
        </p>

        <br />

        <p className='py-2'>5. üåé W-8-EXP</p>

        <p className='py-2'>‚úÖ Not a US Citizen or Resident</p>

        <p className='py-2'>‚úÖ Foreign Government or Other Foreign Organization</p>

        <p className='py-2'>
          Instructions:{' '}
          <Link
            href={'https://www.irs.gov/pub/irs-pdf/iw8exp.pdf'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/pub/irs-pdf/iw8exp.pdf
          </Link>
        </p>

        <p className='py-2'>
          Form:{' '}
          <Link
            href={'https://www.irs.gov/pub/irs-pdf/fw8exp.pdf'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/pub/irs-pdf/fw8exp.pdf
          </Link>
        </p>

        <br />

        <p className='py-2'>6. üåé W-8-IMY</p>

        <p className='py-2'>‚úÖ Not a US Citizen or Resident</p>

        <p className='py-2'>‚úÖ Foreign Intermediary, Foreign Flow-Through Entity, or Certain U.S. Branches</p>

        <p className='py-2'>
          Instructions:{' '}
          <Link
            href={'https://www.irs.gov/individuals/international-taxpayers/form-w-8imy'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/individuals/international-taxpayers/form-w-8imy
          </Link>
        </p>

        <p className='py-2'>
          Form:{' '}
          <Link
            href={'https://www.irs.gov/pub/irs-pdf/fw8imy.pdf'}
            rel='noopener norefferer'
            target='_blank'
            className='text-link-colour hover:underline col-span-2'
          >
            https://www.irs.gov/pub/irs-pdf/fw8imy.pdf
          </Link>
        </p>

        <br />

        <p className='py-2'>
          The IRS requires us to withhold FIL from payments to non-U.S. citizens who live outside of the United States
          unless we can associate the payee with a completed, signed W-8/W-9 form.
        </p>
      </div>
    </ModalLarge>
  );
};

export default index;
