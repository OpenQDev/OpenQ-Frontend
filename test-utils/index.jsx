// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/FundStore/TestFundProvider';
import TokenProvider from '../components/TokenSelection/TokenStore/TokenProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
afterEach(() => {
  cleanup();
});

const customRender = (ui, options = {}, storeProps, authProps) => {
  const IntersectionObserverMock = vi.fn(() => ({
    disconnect: vi.fn(),
    observe: vi.fn(),
    takeRecords: vi.fn(),
    unobserve: vi.fn(),
  }));
  vi.stubGlobal('IntersectionObserver', IntersectionObserverMock);
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <TokenProvider>{children}</TokenProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}>
            <TokenProvider>{children}</TokenProvider>
          </FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
