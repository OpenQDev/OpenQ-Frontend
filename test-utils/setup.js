import { expect, beforeEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import matchers from '@testing-library/jest-dom/matchers';

expect.extend(matchers);
beforeEach(() => {
  cleanup();
});
process.on('unhandledRejection', (reason) => {
  // eslint-disable-next-line no-console
  console.log(`FAILED TO HANDLE PROMISE REJECTION`);
  throw reason;
});
