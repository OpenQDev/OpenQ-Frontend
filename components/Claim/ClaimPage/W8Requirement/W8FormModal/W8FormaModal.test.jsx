import { render } from '../../../../../test-utils';
import React from 'react';
import W8FormModal from '.';

describe('W8FormModal', () => {
  it('should match initial DOM Snapshot', () => {
    const { asFragment } = render(<W8FormModal />);
    expect(asFragment()).toMatchSnapshot();
  });
});
