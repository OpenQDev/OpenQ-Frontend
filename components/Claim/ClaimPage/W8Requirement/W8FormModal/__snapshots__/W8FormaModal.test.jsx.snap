// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`W8FormModal > should match initial DOM Snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex justify-center items-center overflow-x-hidden overflow-y-auto fixed inset-0 z-50 undefined"
    >
      <div
        class="flex w-full md:w-[640px] h-full md:h-[600px] border border-gray-700 md:rounded-sm bg-[#161B22]"
      >
        <div
          class="flex flex-col relative w-full"
        >
          <button
            class="absolute top-4 right-4 cursor-pointer"
            data-testid="cross"
          >
            <svg
              class="fill-muted hover:fill-blue-400 w-4 h-4"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
          <div
            class="py-2 px-4 border-b border-gray-700 text-2xl w-full"
          >
            W8/W9 Form
          </div>
          <div
            class="w-full h-full md:h-[500px] overflow-x-hidden overflow-y-auto"
          >
            <div
              class="whitespace-pre-wrap px-8 py-2"
            >
              <p
                class="py-2"
              >
                Please complete and upload a form W-8. Choose one of five types, depending on your entity. We encourage you to consult with your own tax or financial adviser to determine which form is appropriate for you.
              </p>
              <br />
              <p
                class="flex py-2 gap-2"
              >
                1.
                <img
                  alt="US-flag"
                  data-nimg="1"
                  decoding="async"
                  height="24"
                  loading="lazy"
                  src="/_next/image?url=%2Fdiverse%2Fus-flag.jpg&w=48&q=75"
                  srcset="/_next/image?url=%2Fdiverse%2Fus-flag.jpg&w=32&q=75 1x, /_next/image?url=%2Fdiverse%2Fus-flag.jpg&w=48&q=75 2x"
                  style="color: transparent;"
                  width="24"
                />
                W-9
              </p>
              <p
                class="py-2"
              >
                ‚úÖ US Citizen or Resident
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Claiming as an individual
              </p>
              <p
                class="py-2"
              >
                Instructions: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/forms-pubs/about-form-w-9"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/forms-pubs/about-form-w-9
                </a>
              </p>
              <p
                class="py-2"
              >
                Form: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/pub/irs-pdf/fw9.pdf"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/pub/irs-pdf/fw9.pdf
                </a>
              </p>
              <br />
              <p
                class="py-2"
              >
                2. üåç W-8-BEN
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Not a US Citizen or Resident
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Claiming as an individual
              </p>
              <p
                class="py-2"
              >
                Instructions: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/instructions/iw8ben"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/instructions/iw8ben
                </a>
              </p>
              <p
                class="py-2"
              >
                Form: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/pub/irs-pdf/fw8ben.pdf"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/pub/irs-pdf/fw8ben.pdf
                </a>
              </p>
              <br />
              <p
                class="py-2"
              >
                3. üåé W-8-BEN-E
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Not a US Citizen or Resident
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Claiming as an entity
              </p>
              <p
                class="py-2"
              >
                Instructions: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/instructions/iw8bene"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/instructions/iw8bene
                </a>
              </p>
              <p
                class="py-2"
              >
                Form: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/pub/irs-pdf/fw8bene.pdf"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/pub/irs-pdf/fw8bene.pdf
                </a>
              </p>
              <br />
              <p
                class="py-2"
              >
                4. üåé W-8-ECI
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Not a US Citizen or Resident
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Foreign Person‚Äôs Claim That Income is Effectively Connected With the Conduct of a Trade or Business in the United States
              </p>
              <p
                class="py-2"
              >
                Instructions: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/pub/irs-pdf/iw8eci.pdf"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/pub/irs-pdf/iw8eci.pdf
                </a>
              </p>
              <p
                class="py-2"
              >
                Form: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/pub/irs-pdf/fw8eci.pdf"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/pub/irs-pdf/fw8eci.pdf
                </a>
              </p>
              <br />
              <p
                class="py-2"
              >
                5. üåé W-8-EXP
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Not a US Citizen or Resident
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Foreign Government or Other Foreign Organization
              </p>
              <p
                class="py-2"
              >
                Instructions: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/pub/irs-pdf/iw8exp.pdf"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/pub/irs-pdf/iw8exp.pdf
                </a>
              </p>
              <p
                class="py-2"
              >
                Form: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/pub/irs-pdf/fw8exp.pdf"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/pub/irs-pdf/fw8exp.pdf
                </a>
              </p>
              <br />
              <p
                class="py-2"
              >
                6. üåé W-8-IMY
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Not a US Citizen or Resident
              </p>
              <p
                class="py-2"
              >
                ‚úÖ Foreign Intermediary, Foreign Flow-Through Entity, or Certain U.S. Branches
              </p>
              <p
                class="py-2"
              >
                Instructions: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/individuals/international-taxpayers/form-w-8imy"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/individuals/international-taxpayers/form-w-8imy
                </a>
              </p>
              <p
                class="py-2"
              >
                Form: 
                <a
                  class="text-link-colour hover:underline col-span-2"
                  href="https://www.irs.gov/pub/irs-pdf/fw8imy.pdf"
                  rel="noopener norefferer"
                  target="_blank"
                >
                  https://www.irs.gov/pub/irs-pdf/fw8imy.pdf
                </a>
              </p>
              <br />
              <p
                class="py-2"
              >
                The IRS requires us to withhold FIL from payments to non-U.S. citizens who live outside of the United States unless we can associate the payee with a completed, signed W-8/W-9 form.
              </p>
            </div>
          </div>
          <div
            class="flex justify-between items-center py-2 px-4 flex-wrap border-t border-gray-700 w-full "
          >
            <div
              class="my-1"
            />
            <div>
              <button
                class="btn-primary"
              >
                I understand
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="bg-overlay z-10 fixed inset-0"
    />
  </div>
</DocumentFragment>
`;
